### ISSUE TITLE
Ruff incorrectly reports on redefined imports with explicit re-exports

### ISSUE THREAD

**Issue by charliermarsh** [charliermarsh/ruff#1043]
Given this snippet:

```py
import os
import os as os
```

Ruff reports an unused import on line 2, although that's an explicit re-export:

```py
∴ cargo run foo.py
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `target/debug/ruff foo.py`
Found 1 error(s).
foo.py:2:8: F401 `os` imported but unused
```

**Comment by charliermarsh** [comment]
It's easy to fix:

```py
import os
import os as os
```

But harder to fix:

```py
import os as os
import os
```

Fixing the latter requires that we can detect redefinition (see: #996).

**Comment by charliermarsh** [comment]
Oh, this works now.
